libs/logging/src/main/java/org/elasticsearch/logging/spi/LogManagerFactory.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.logging.spi;
+
+import org.elasticsearch.logging.Logger;
+
+import java.util.ServiceLoader;
+
+/**
+ * SPI for creating new loggers
+ */
+public interface LogManagerFactory {
+
+    LogManagerFactory INSTANCE = loadProvider();
+
+    static LogManagerFactory loadProvider() {
+        ServiceLoader<LogManagerFactory> sl = ServiceLoader.load(LogManagerFactory.class, ClassLoader.getSystemClassLoader());
```

Comment:
I was thinking that this class will change when we move the log4j off the system classloader. Something to what we have here https://github.com/ChrisHegarty/elasticsearch/blob/logging-module/libs/logging/src/main/java/org/elasticsearch/logging/spi/LogManagerFactory.java

Reply:
How about we look up the implementation using the service loader variant that accepts a module layer? Or is the question more related to the use of service loader at all?

Reply:
The latter. While eventually I think service loader will be used when we can move log4j off the system classloader, I was suggesting starting by loading this in server and setting statically here.

Reply:
What is the benefit of setting this statically from server? We would have to remember to set this with LogConfigurator, and once we move implementation off the system classloader to remove it. 
I personally think this is an easier to read solution, but don't have strong opinion against setting as well.

Reply:
For stable plugin api, we are thinking about changing the way the system classloader is setup. Right now, everything from server is in the system classloader. For stable plugin api, we want to first load the stable api jars, then create a separate loader, parented by the stable api jars, that is everything from server. That separation would break this ServerLoader lookup, because the logging api impl would not be on the system classloader, and this code could not statically find the loader wtih the impl (since it will not yet have been created). Thus, setting this statically as an init of the logging api allows us to delay until server has actually been loaded.